pkg/virtualization/core/virtualsystem/virtualmachine.go

```diff
@@ -772,79 +772,106 @@ func (vm *VirtualMachine) GetDvdDrives() (col drive.DvdDriveCollection, err erro
+			if err1 != nil {
+				// Missing related storage allocation data is benign
+				// It just means that DVD disk is not available. No need to return error
 
-		dvdPath := ""
-		for _, interfaceValue := range dvdPathInterface.([]interface{}) {
-			valuetmp, ok := interfaceValue.(string)
-			if !ok {
-				err = errors.Wrapf(errors.InvalidType, " string is Invalid. Expected %s", reflect.TypeOf(interfaceValue))
+				// since we are in an anonymus function, we don't want to return an error here.
+				// we just want to continue to the next dvd drive
+				err1 = nil
 				return
 			}
+			defer logicalDiskInstance.Close()
 
-			dvdPath = string(valuetmp)
-		}
+			dvdPathInterface, err1 := logicalDiskInstance.GetProperty("HostResource")
+			if err1 != nil {
+				err1 = fmt.Errorf("unable to read HostResource field from disk WMI %s", err1)
+				return
+			}
+			if dvdPathInterface == nil {
+				// Attached ISO path is not available, continue
+				return
+			}
 
-		if filepath.Clean(dvdPath) != filepath.Clean(isoPath) {
-			continue
-		}
+			dvdPath := ""
+			for _, interfaceValue := range dvdPathInterface.([]interface{}) {
+				valuetmp, ok := interfaceValue.(string)
+				if !ok {
+					err1 = errors.Wrapf(errors.InvalidType, " string is Invalid. Expected %s", reflect.TypeOf(interfaceValue))
+					return
+				}
 
-		// Clone the DVD drive instance for return
-		dvdclone, err1 := inst.Clone()
-		if err1 != nil {
-			err = err1
-			return
-		}
+				dvdPath = string(valuetmp)
+			}
 
-		retdvd, err1 := drive.NewDvdDrive(dvdclone)
-		if err1 != nil {
-			err = err1
-			return
-		}
+			if filepath.Clean(dvdPath) != filepath.Clean(isoPath) {
+				return
+			}
 
-		// Also clone the logical disk instance for return
-		dvddiskclone, err1 := dvddisk.Clone()
-		if err1 != nil {
-			err = err1
+			// TODO (pgayam): Do we need to clone the dvdDriveInstance and logicalDiskInstance?
+			//Can we directly use them to create dvdDrive and logicalDisk?
+
+			// Clone the DVD drive instance for return
+			dvdDriveclone, err1 := dvdDriveInstance.Clone()
+			if err1 != nil {
+				return
+			}
+			defer dvdDriveclone.Close()
+
+			dvdDrive, err1 = drive.NewDvdDrive(dvdDriveclone)
+			if err1 != nil {
+				// no need to close dvdDrive if error is returned
+				return
+			}
+
+			// Also clone the logical disk instance for return
+			logicalDiskClone, err1 := logicalDiskInstance.Clone()
+			if err1 != nil {
+				return
+			}
+			defer logicalDiskClone.Close()
+
+			logicalDisk, err1 = disk.NewLogicalDisk(logicalDiskClone)
+			if err != nil {
+				// no need to close logicalDisk if error is returned
+				return
+			}
+
+			success = true
 			return
-		}
+		}()
 
-		retdvddisk, err1 := disk.NewLogicalDisk(dvddiskclone)
-		if err1 != nil {
-			err = err1
+		if err != nil || succuess {
 			return
 		}
+	}
 
-		dvd = retdvd
-		diskdvd = retdvddisk
-		return
+	// If we are here, we didn't find the DVD drive with the given ISO path
```

Comment:
Can you install the below as defer in the beginning of the function to keep the intention clean